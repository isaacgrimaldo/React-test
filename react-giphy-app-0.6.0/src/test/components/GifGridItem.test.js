import React from 'react';
import'@testing-library/jest-dom';
import {shallow} from 'enzyme'; 

import {GifGridItem} from '../../components/GifGridItem';

describe('Pruebad del documento GifGridItem.js', () => {
    
    const title = 'hola';
    const url   = 'http://localhost:8080/images.png';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);
    
    test('Debe mostrar correctamente el HTML con su propiedades', () => {
       expect(wrapper).toMatchSnapshot();
     })
    
     test('El title debé de mostrar correctamente', () => {
         const p = wrapper.find('p').text().trim();
         expect(p).toBe(title);
     })

     test('propiedades de img debén mostrase correctamente', () => {
         const img = wrapper.find('img');
         expect(img.prop('src')).toBe(url);
         expect(img.prop('alt')).toBe(title);         
     })

     test('El contenedor de la imagen debe tener la classe animate__fadeIn', () => {
        const div = wrapper.find('div');
        expect(div.hasClass('animate__fadeIn')).toBe(true);
     });

});