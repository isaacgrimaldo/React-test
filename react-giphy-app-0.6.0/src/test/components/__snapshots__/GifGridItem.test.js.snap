// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebad del documento GifGridItem.js Debe mostrar correctamente el HTML con su propiedades 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="hola"
    src="http://localhost:8080/images.png"
  />
  <p>
     
    hola
     
  </p>
</div>
`;
